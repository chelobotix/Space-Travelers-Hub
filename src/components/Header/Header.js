import NavBar from "./Navbar";
import style from './Header.module.css'

const Header = () => (
  <header className={style.header}>
    <NavBar />
  </header>
);

export default Header
