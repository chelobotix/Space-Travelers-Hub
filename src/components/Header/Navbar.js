import { NavLink } from "react-router-dom";
import planetImg from "../../assets/images/planet.png"

const NavBar = () => {
  return (
    <nav>
      <ul>
        <li>
          <NavLink
            to="/rockets"
            style={({ isActive }) => (isActive ? { color: "red" } : {})}
          >
            <img src={planetImg} alt="" />
            <p>Space Travelers' Hub</p>
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/rockets"
            style={({ isActive }) => (isActive ? { color: "red" } : {})}
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/missions"
            style={({ isActive }) => (isActive ? { color: "red" } : {})}
          >
            Missions
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/profile"
            style={({ isActive }) => (isActive ? { color: "red" } : {})}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  )
}

export default NavBar;



