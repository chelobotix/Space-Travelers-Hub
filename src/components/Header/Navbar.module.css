.nav {
  width: 100%;
}

.hr {
  border-top: 1px solid rgb(226, 222, 216);
}

.ul {
  display: flex;
  align-items: center;
  list-style-type: none;
  padding: 0 10px;
  margin: 0;
  margin-bottom: 30px;
}

.logo {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: #000;
  gap: 10px;
}

.img {
  width: 60px;
}

.logoName {
  font-size: 2.4rem;
  margin: 10px;
}

.right {
  margin-left: auto;
}

.normalLink {
  color: #54a0f8;
  padding: 5px 10px;
}

.border {
  border-left: 1px solid #000;
}
