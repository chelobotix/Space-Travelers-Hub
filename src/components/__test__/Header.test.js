import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../Header/Header';

describe('Header', () => {
  test('Check if render', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const header = screen.getByRole('banner');
    expect(header).toBeInTheDocument();
  });

  test('Header Tree', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
