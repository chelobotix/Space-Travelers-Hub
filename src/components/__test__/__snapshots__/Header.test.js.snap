// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Tree 1`] = `
<header
  className="header"
>
  <nav
    className="nav"
  >
    <ul
      className="ul"
    >
      <li>
        <a
          className="logo"
          href="/rockets"
          onClick={[Function]}
        >
          <img
            alt=""
            className="img"
            src="planet.png"
          />
          <p
            className="logoName"
          >
            Space Travelers' Hub
          </p>
        </a>
      </li>
      <li
        className="right"
      >
        <a
          className="normalLink"
          href="/rockets"
          onClick={[Function]}
          style={Object {}}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="normalLink"
          href="/missions"
          onClick={[Function]}
          style={Object {}}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="normalLink border"
          href="/profile"
          onClick={[Function]}
          style={Object {}}
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
